src/sulogin.c

```diff
@@ -202,10 +203,10 @@ static void catch_signals (unused int sig)
 			execl (PATH_TELINIT, "telinit", RUNLEVEL, (char *) 0);
 #endif
 			exit (0);
-		} else {
-			STRFCPY (pass, cp);
-			strzero (cp);
 		}
+		STRFCPY (pass, cp);
+		erase_pass (cp);
+
```

Comment:
BTW, I maybe should have splitted this into a separate commit:

Don't `else` after a `noreturn` call (`exit(3)`).

Reply:
I will split it.  Definitely.  I don't want someone auditing the commits in the future and think "why the hell did this change?  no commit message at all about it...".